packages/indexer/lib/index.js

```diff
@@ -207,28 +207,47 @@ class TransactionCollector {
   async *collect() {
     let hashes = new Set();
     if (this.lock) {
-      for (const h of this.indexer.getTransactionsByLockScript(this.lock, {
+      const iter = this.indexer.getTransactionsByLockScriptIterator(this.lock, {
         validateFirst: false,
-      })) {
-        hashes = hashes.add(h);
+      });
+      while (true) {
+        const hash = iter.next();
+        if (!hash) {
+          break;
+        }
+        hashes = hashes.add(hash);
+        const tx = await this.rpc.get_transaction(hash);
+        if (!this.skipMissing && !tx) {
+          throw new Error(`Transaction ${h} is missing!`);
+        }
+        if (this.includeStatus) {
+          yield tx;
+        } else {
+          yield tx.transaction;
+        }
       }
     }
     if (this.type) {
-      for (const h of this.indexer.getTransactionsByTypeScript(this.type, {
+      const iter = this.indexer.getTransactionsByTypeScriptIterator(this.type, {
         validateFirst: false,
-      })) {
-        hashes = hashes.add(h);
-      }
-    }
-    for (const h of hashes) {
-      const tx = await this.rpc.get_transaction(h);
-      if (!this.skipMissing && !tx) {
-        throw new Error(`Transaction ${h} is missing!`);
-      }
-      if (this.includeStatus) {
-        yield tx;
-      } else {
-        yield tx.transaction;
+      });
+      while (true) {
+        const hash = iter.next();
+        if (!hash) {
+          break;
+        }
+        if (hashes.has(hash)) {
```

Comment:
Forget to add hashes ?

Reply:
Here we don't have to insert the hashes. The set is only enabling us to skip those transactions that already have matched lock scripts. Querying only based on type scripts here, won't result in duplicate items.

Reply:
OK, I misunderstood before.